pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -290,6 +314,51 @@ impl Layer for DeltaLayer {
         Ok(())
     }
 
+    fn get_occupied_ranges(&self, ctx: &RequestContext) -> Result<Vec<Range<Key>>> {
+        if let Ok(inner) = self.load(ctx) {
```

Comment:
Shouldn't there be a `?` here, or at least log the error ... feels strange it's just ignored. At least add some justification comment for ignoring it?

Reply:
This is ugly workaround I have to add to make 3 tests passed. Otherwise errors while invalid (specially corrupted) layer loading cause failure in the different place then test expected.

Reply:
I think this in the center of needing to add `ctx` around in LayerMap. Could there be an another way of always making sure we have the correct information available for layers in the LayerMap..

This comment could also be part of: https://github.com/neondatabase/neon/pull/3348/files#r1098257312

Reply:
> Could there be an another way of always making sure we have the correct information available for layers in the LayerMap

I will be very happy if there is such way. But I do not know one except TLS. And looks like nobody likes it.